processingthetext.py

```diff
@@ -0,0 +1,191 @@
+
+count_character_gender(char_dict)
+
+
+def handle_2word_chars(list_of_words):
+    '''Deals with all 2-word character abbreviations in the script.
+    i.e. 'M.' becomes 'M. Wife.' like it appears in the dictionary.
+        NOTE: Does not account for 'Cap. Wife' (because 'Cap.' is a
+        different character so the method of combining any string
+        that starts with 'Cap.' with the string that comes after it
+        would not increase overall accuracy.)'''
+    for i in range(len(list_of_words)):
+        if list_of_words[i] in ['M.', '1.', '2.', '3.']:            # if the word IS in list of words
+            list_of_words[i] = list_of_words[i] + str(' ') + list_of_words[i+1]
+    return list_of_words
+
+
+def parse(abbr_character_name,text_file_name):
+    '''Takes abbreviated character name as input (ie "Samp." for Sampson)
+    Counts and returns # of times each abbreviated name appears (and
+    therefore the number of times the character speaks in the text).
+        NOTE: 'Cap. Wife.' and 'Wife.' are the same person but this script
+        is not currently aware of that fact (which is why it thinks there
+        are 5 female characters when in reality there are 4. This also
+        causes discrepancies in the final calculation of the averages.)'''
+    number_of_mentions = 0
+    for word in text_file_name:
+        if word == abbr_character_name:
+            number_of_mentions = number_of_mentions + 1
+    return number_of_mentions
+
+
+modified_text = handle_2word_chars(list_of_words)
+# print(modified_text)                  # unit test; returns text with 2-word names in single strings
+
+# parse('1. Serv.',modified_text)       # unit test; returns number of mentions for given character name
+
+
+def parse_text_for_mentions(char_dict, text_file_name):
+    '''Counts number of times each key of dict_name appears in modified_text.
+    Takes a dictionary that maps character name to gender and a text file (in the form of a list of strings)
+    Returns name_number: a dictionary that maps character names to number of times they speak.'''
+    mentions = {}
+    for key in char_dict.keys():
+        number_of_mentions = parse(key, text_file_name)
+        mentions[key] = number_of_mentions
+    return mentions
+
+parse_text_for_mentions(char_dict, modified_text)
+name_number = dict(parse_text_for_mentions(char_dict,modified_text))
+
+
+# print(name_number)         # unit test; returns dictionary with character names as keys and # of times mentioned as value
+
+
+def average_times_speaking(name_number,char_dict):
+    '''Calculates average number of times male and female characters speak.
+    Takes 2 dictionaries as input arguments:
+        1. name_number: maps character names to how many times they speak
+        2. char_dict: maps character names to gender
+    Returns average number of times characters of each gender speak. '''
+    for key,value in name_number.items():
+        name_number[key] = (char_dict[key],value)
```

Comment:
This for loop is not intuitive or simple enough for a reader to quickly understand what it does. Comments on what it does would be helpful.